src/controllers/reset-password.js

```
@@ -0,0 +1,83 @@
+const { scryptSync } = require('crypto');
+const jwt = require('jsonwebtoken');
+const { cryptoLink, decryptLink } = require('./crypto');
+// eslint-disable-next-line no-useless-escape
+const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/;
+const setResetPassword = async (req, res) => {
+  try {
+    const schema = Joi.object({ email: Joi.string().email().required() });
+    const { error } = schema.validate(req.body);
+    if (error) return res.status(201).send(error.details[0].message);
+
+    const user = await User.findOne({ where: { email: req.body.email } });
+    if (!user)
+      return res
+        .status(201)
+        .send(
+          'Sorry we can’t find an account with the entered email address. Please try again or create new account.'
+        );
+
+    const token = {
+      userId: user.id,
+      token: jwt.sign({ email: req.body.email }, 'Stack', {
+        expiresIn: '1800s',
+      }),
+    };
+    const encryptedToken = cryptoLink(token.token);
+    // eslint-disable-next-line max-len
+    const link = `http://reverifi-api.docker/api/password-reset/${user.dataValues.id}/${encryptedToken}`;
+    res.send(link);
+  } catch (error) {
+    res.send('An error occurred');
+  }
+};
+const hashSalt = 'hash-salt';
+
+const getHash = (password) => scryptSync(password, hashSalt, 32).toString('hex');
+
+const setUserToken = async (req, res) => {
+  const decryptedToken = decryptLink(req.params.token);
+  const isTokenExpired = (token) =>
+    Date.now() >= JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp * 1000;
+  try {
+    const schema = Joi.object({
+      password: Joi.string().required().regex(passwordRegex),
+      confirmPassword: Joi.string().required().valid(Joi.ref('password')),
+    });
+    const user = await User.findOne({ where: { id: req.params.userId } });
+    if (!user || isTokenExpired(decryptedToken))
+      return res.status(201).send('invalid link or expired');
```

you can throw an error here